tasks/main.yml

```diff
@@ -1 +1,30 @@
 # role tasks
+- name: Rebooting in {{ reboot_seconds_before_reboot }} seconds
+  command: >
+    systemd-run --on-active="{{ reboot_seconds_before_reboot }}"
+    systemctl reboot
+  become: "{{ reboot_become }}"
+  become_user: "{{ reboot_become_user }}"
+  tags: skip_ansible_lint
+
+- name: >
+    Reboot command has been issued, waiting a maximum of
+    {{ reboot_wait_for_connection_timeout }} seconds to reconnect.
+  wait_for_connection:
+    delay: "{{ reboot_wait_for_connection_delay }}"
+    timeout: "{{ reboot_wait_for_connection_timeout }}"
+
+- name: Get the system uptime before reboot
+  set_fact:
+    reboot_before: "{{ ansible_uptime_seconds }}"
+
+- name: Get the system uptime after reboot
+  setup:
+    gather_subset: '!all,!min,hardware'
+
+- name: Provide feedback on failure
+  fail:
+    msg: >
+      "Failed to reboot.  Uptime did not decrease.  Previous uptime
+      {{ reboot_before }} Post uptime {{ reboot_ansible_uptime_seconds }}"
+  when: reboot_before <= ansible_uptime_seconds
```

Comment:
- This is perhaps a personal stylistic thing, but using the `fail` task here means that users will always see a "Provide feedback on failure" task, even if there was no failure. This is effectively an assertion, for which ansible has a task:
```yaml
- name: Assert that reboot succeeded based on system uptime
  assert:
    that: "reboot_before < ansible_uptime_seconds"
    msg: >
      Failed to reboot:  Uptime did not decrease.
```
- I don't personally think there's value in including the previous and current uptime values in the message, since there's nothing a user can really do with them.

- Is there a reason that `<=` is used in the failure assertion/when conditional? The failure message indicates `<`, so that's what I expected to see there.

- Note that using the line-folding syntax (`>`) means that the double quotes in `msg` become redundant.

Given the following playbook...:
```yaml
- name: line folding test
  hosts: localhost
  vars:
    testvar: quotes
  tasks:
    - debug:
        msg: >
          "This is a line-folded message
           with {{ testvar }} included."
    - debug:
        msg: >
          This is a line-folded message
          with {{ testvar }} excluded.
```

...here's what ansible outputs:
```
TASK [debug] ***
ok: [localhost] => 
  msg: |-
    "This is a line-folded message
     with quotes included."

TASK [debug] ***
ok: [localhost] => 
  msg: |-
    This is a line-folded message with quotes excluded.
```

`{{ testvar }}` is just in there to demonstrate that variable expansion still works as expected.